Failing ScheduledExecutorServiceBasicTest.scheduled_executor_collects_statistics_when_stats_enabled

The test `ScheduledExecutorServiceBasicTest.scheduled_executor_collects_statistics_when_stats_enabled` failed on PR builder (`master` branch).
http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/2689/

**Stacktrace**
```
java.lang.AssertionError: Expected 'totalExecutionTime::{totalExecutionTime=[0, 1003], creationTime=[1602678490366, 1602678490366], totalStartLatency=[0, 1], pending=[0, 0], cancelled=[0, 0], started=[1, 1], completed=[0, 1]}' to be greater than or equal to 1000, but was 0
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at com.hazelcast.test.HazelcastTestSupport.assertGreaterOrEquals(HazelcastTestSupport.java:1401)
	at com.hazelcast.scheduledexecutor.impl.ScheduledExecutorServiceBasicTest.assertMetricsCollected(ScheduledExecutorServiceBasicTest.java:1719)
	at com.hazelcast.scheduledexecutor.impl.ScheduledExecutorServiceBasicTest.scheduled_executor_collects_statistics_when_stats_enabled(ScheduledExecutorServiceBasicTest.java:1596)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/hazelcast/src/test/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorServiceBasicTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorServiceBasicTest.java
@@ -1589,12 +1589,14 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
             entry.getValue().get();
         }
 
-        // collect metrics
-        Map<String, List<Long>> metrics = collectMetrics(SCHEDULED_EXECUTOR_PREFIX, instances);
+        assertTrueEventually(() -> {
+            // collect metrics
+            Map<String, List<Long>> metrics = collectMetrics(SCHEDULED_EXECUTOR_PREFIX, instances);
 
-        // check results
-        assertMetricsCollected(metrics, 1000, 0,
+            // check results
+            assertMetricsCollected(metrics, 1000, 0,
                 1, 1, 0, 1, 0);
+        });
     }
 
     @Test
@@ -1612,11 +1614,13 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
             entry.getValue().get();
         }
 
-        // collect metrics
-        Map<String, List<Long>> metrics = collectMetrics(SCHEDULED_EXECUTOR_PREFIX, instances);
+        assertTrueAllTheTime(() -> {
+            // collect metrics
+            Map<String, List<Long>> metrics = collectMetrics(SCHEDULED_EXECUTOR_PREFIX, instances);
 
-        // check results
-        assertTrue("No metrics collection expected but " + metrics, metrics.isEmpty());
+            // check results
+            assertTrue("No metrics collection expected but " + metrics, metrics.isEmpty());
+        }, 5);
     }
 
     @Test
@@ -1635,13 +1639,15 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
 
         assertOpenEventually(progress);
 
-        // collect metrics
-        Map<String, List<Long>> metrics = collectMetrics(SCHEDULED_EXECUTOR_PREFIX, instances);
-
-        // check results
         try {
-            assertMetricsCollected(metrics, 0, 0,
+            assertTrueEventually(() -> {
+                // collect metrics
+                Map<String, List<Long>> metrics = collectMetrics(SCHEDULED_EXECUTOR_PREFIX, instances);
+
+                // check results
+                assertMetricsCollected(metrics, 0, 0,
                     3, 2, 0, now, 0);
+            });
         } finally {
             suspend.release();
         }
@@ -1663,11 +1669,13 @@ public class ScheduledExecutorServiceBasicTest extends ScheduledExecutorServiceT
 
         assertOpenEventually(progress);
 
-        // collect metrics
-        Map<String, List<Long>> metrics = collectMetrics(SCHEDULED_EXECUTOR_PREFIX, instances);
+        assertTrueAllTheTime(() -> {
+            // collect metrics
+            Map<String, List<Long>> metrics = collectMetrics(SCHEDULED_EXECUTOR_PREFIX, instances);
 
-        // check results
-        assertTrue("No metrics collection expected but " + metrics, metrics.isEmpty());
+            // check results
+            assertTrue("No metrics collection expected but " + metrics, metrics.isEmpty());
+        }, 5);
     }
 
 

